include/boost/beast/_experimental/test/impl/stream.hpp

```diff
@@ -44,7 +39,8 @@ class basic_stream<Executor>::read_op : public detail::stream_read_op_base
 #if defined(BOOST_ASIO_NO_TS_EXECUTORS)
         net::any_io_executor wg2_;
 #else // defined(BOOST_ASIO_NO_TS_EXECUTORS)
-        net::executor_work_guard<ex2_type> wg2_;
+        net::executor_work_guard<
+            net::associated_executor_t<Handler, net::any_io_executor>> wg2_;
```

Comment:
@klemens-morgenstern Here we use any_io_executor when there is no associated executor because it will be initialized with s->exec (when there is no associated) in the following line: https://github.com/boostorg/beast/blob/35d533299c129d622af774f96b820e8f18df068f/include/boost/beast/_experimental/test/impl/stream.hpp#L68

`stream_state::exec` has an executor type of any_io_executor:
https://github.com/boostorg/beast/blob/develop/include/boost/beast/_experimental/test/detail/stream_state.hpp#L88

This was causing compile time error in instantiation of `is_async_read_stream` type trait, because the handler has no associated executor thus it tries to initialize `wg2_` with an instance of `s->exec`. which wg2_ has an executor type of io_context::basic_executor_type.

Reply:
I don't think this is correct. If you don't haven associated one shouldn't you default to `ex1_type`?

Reply:
`ex1_type` is `basic_stream<Executor>::executor_type` but we are initialization the `wg2_` with `s->exec` which is `any_io_executor`.
Just as it had been done for `wg1_`:
https://github.com/boostorg/beast/blob/35d533299c129d622af774f96b820e8f18df068f/include/boost/beast/_experimental/test/impl/stream.hpp#L133
https://github.com/boostorg/beast/blob/35d533299c129d622af774f96b820e8f18df068f/include/boost/beast/_experimental/test/impl/stream.hpp#L146